src/layers/base-layer.js

```diff
@@ -297,7 +297,15 @@ export default class Layer {
       sizeScale: 'linear',
       sizeField: null,
 
-      visConfig: {}
+      visConfig: {},
+
+      textLabel: {
+        field: null,
+        color: [255, 255, 255],
```

Comment:
I'm not sure color and size should be a constant value here. I imagine the color and size also can be mapped to a field ?